src/providers/base_javascript.js

```diff
@@ -1,212 +1,339 @@
+   * @returns {Map<string, PackageURL>} Map of dependency names to their PackageURL objects
+   * @protected
+   */
+	_getRootDependencies(depTree) {
+		if (!depTree.dependencies) {
+			return new Map();
+		}
+
+		return new Map(
+			Object.entries(depTree.dependencies).map(
+				([key, value]) => [key, toPurl(ecosystem, key, value.version)]
+			)
+		);
+	}
+
+	/**
+   * Executes the list command to get dependencies
+   * @param {boolean} includeTransitive - Whether to include transitive dependencies
+   * @param {string} manifestDir - The manifest directory
+   * @returns {string} The command output
+   * @private
+   */
 	#executeListCmd(includeTransitive, manifestDir) {
 		const listArgs = this._listCmdArgs(includeTransitive, manifestDir);
 		return this.#invokeCommand(listArgs);
 	}
 
-	#version() {
-		this.#invokeCommand(['--version'], { stdio: 'ignore' });
+	/**
+   * Gets the version of the package manager
+   * @returns {string} The version string of the package manager
+   * @protected
+   */
+	_version() {
+		return this.#invokeCommand(['--version']);
 	}
 
+	/**
+   * Creates or updates the lock file for the package manager
+   * @param {string} manifestDir - Directory containing the manifest file
+   * @private
+   */
 	#createLockFile(manifestDir) {
-		// in windows os, --prefix flag doesn't work, it behaves really weird , instead of installing the package.json fromm the prefix folder,
-		// it's installing package.json (placed in current working directory of process) into prefix directory, so
-		let originalDir = process.cwd()
-		if (os.platform() === 'win32') {
-			process.chdir(manifestDir)
+		const originalDir = process.cwd();
+		const isWindows = os.platform() === 'win32';
+
+		if (isWindows) {
+			// On Windows, --prefix flag doesn't work as expected
+			// Instead of installing from the prefix folder, it installs from current working directory
+			process.chdir(manifestDir);
 		}
-		const args = this._updateLockFileCmdArgs(manifestDir);
+
 		try {
+			const args = this._updateLockFileCmdArgs(manifestDir);
 			this.#invokeCommand(args);
 		} finally {
-			if (os.platform() === 'win32') {
-				process.chdir(originalDir)
+			if (isWindows) {
+				process.chdir(originalDir);
 			}
 		}
 	}
 
+	/**
+   * Invokes a command with the given arguments
+   * @param {string[]} args - Command arguments
+   * @param {Object} [opts={}] - Optional configuration options
+   * @returns {string} Command output
+   * @throws {Error} If command execution fails or command is not found
+   * @private
+   */
 	#invokeCommand(args, opts = {}) {
 		try {
+			if(!opts.cwd) {
+				opts.cwd = path.dirname(this.#manifest.manifestPath);
+			}
 			return invokeCommand(this.#cmd, args, opts);
 		} catch (error) {
 			if (error.code === 'ENOENT') {
-				throw new Error(`${this.#cmd} is not accessible`);
+				throw new Error(`${this.#cmd} is not accessible. Please ensure it is installed and available in your PATH.`);
```

Comment:
Note `#cmd` may be an absolute  path (not looked up via PATH), in case you want to adjust the message to account for this (not necessary, just a note)